### PR TITLE
try fixing travis dot insert into state root debug

### PR DESCRIPTION
**Pull request by winsvega**
this might fix bug with inserting a "." character into stateroot output

@cdetrio @holiman

### CODE REVIEW

File: libevm/VM.cpp [hunk 1]
```diff
@@ -1236,6 +1236,7 @@ void VM::interpretCases()
 
 		CASE(RETURNDATACOPY)
 		{
+			ON_OP();
```
Comment (pirapira) [inline libevm/VM.cpp]: Is this change related to the "inserting dot" bug?

File: test/tools/libtesteth/ImportTest.cpp [hunk 2]
```diff
@@ -226,8 +226,7 @@ std::tuple<eth::State, ImportTest::ExecOutput, eth::ChangeLog> ImportTest::execu
 			st.setShowMnemonics();
 			st.setOptions(Options::get().jsontraceOptions);
 			out = initialState.execute(_env, *se.get(), _tr, Permanence::Committed, st.onOp());
-			cout << st.json();
-			cout << "{\"stateRoot\": \"" << initialState.rootHash().hex() << "\"}";
+			cout << st.json() + "{\"stateRoot\": \"" + initialState.rootHash().hex() + "\"}";
```
Comment (pirapira) [inline test/tools/libtesteth/ImportTest.cpp]: Is this change necessary?

File: test/tools/libtesteth/boostTest.cpp [hunk 3]
```diff
@@ -154,19 +154,31 @@ int main( int argc, char* argv[] )
 			ts1->add(BOOST_TEST_CASE(&customTestSuite));
 			framework::master_test_suite().add(ts1);
 		}
+
+		int result;
```
Comment (pirapira) [inline test/tools/libtesteth/boostTest.cpp]: Initialize it with zero or something (to be consistent with other parts of the codebase).

File: test/tools/libtesteth/boostTest.cpp [hunk 4]
```diff
@@ -154,19 +154,31 @@ int main( int argc, char* argv[] )
...
+		auto fakeInit = [](int, char*[]) -> boost::unit_test::test_suite* { return nullptr; };
+		if (opt.jsontrace || opt.vmtrace || opt.statediff)
+		{
+			// Do not uses travis '.' output thread if debug is defined
```
Comment (pirapira) [inline test/tools/libtesteth/boostTest.cpp]: An extra `s` in `Do not uses`.

File: test/tools/libtesteth/boostTest.cpp [hunk 5]
```diff
@@ -154,19 +154,31 @@ int main( int argc, char* argv[] )
...
+		{
+			// Initialize travis '.' output thread for log activity
+			std::atomic_bool stopTravisOut{false};
+			std::thread outputThread(travisOut, &stopTravisOut);
+			result = unit_test_main(fakeInit, argc, argv);
+			stopTravisOut = true;
+			outputThread.join();
+			dev::test::TestOutputHelper::get().printTestExecStats();
+			return result;
+		}
```
Comment (pirapira) [inline test/tools/libtesteth/boostTest.cpp]: Why is this

File: test/tools/libtesteth/boostTest.cpp [hunk 6]
```diff
@@ -154,19 +154,31 @@ int main( int argc, char* argv[] )
...
 	}
-
-	std::atomic_bool stopTravisOut{false};
-	std::thread outputThread(travisOut, &stopTravisOut);
-	auto fakeInit = [](int, char*[]) -> boost::unit_test::test_suite* { return nullptr; };
-	int result = unit_test_main(fakeInit, argc, argv);
-	stopTravisOut = true;
-	outputThread.join();
-	dev::test::TestOutputHelper::get().printTestExecStats();
-	return result;
```
Comment (pirapira) [inline test/tools/libtesteth/boostTest.cpp]: Why has everything moved into the `try {...}` block?

File: libevm/VM.cpp (hunk 1 above)
Comment (winsvega) [reply libevm/VM.cpp]: @holiman  suggested adding this into this PR. it is related to fixing debug output

File: test/tools/libtesteth/ImportTest.cpp (hunk 2 above)
Comment (winsvega) [reply test/tools/libtesteth/ImportTest.cpp]: no. if we disabled travis out thread it should be fine without this.

File: libevm/VM.cpp (hunk 1 above)
Comment (winsvega) [reply libevm/VM.cpp]: @cdetrio  should test it

Comment (holiman) [reply libevm/VM.cpp]: I'll test it

Comment (pirapira) [reply libevm/VM.cpp]: Would this change also affect Remix?  @yann300 ?

Comment (yann300) [reply libevm/VM.cpp]: That will change the VM trace output yes but I think this change does not require to update Remix
